zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -390,20 +399,12 @@ mod tests {
             db::TABLE_UTXOS,
         ];
 
-        let mut tables_query = st
-            .wallet()
-            .conn
-            .prepare("SELECT sql FROM sqlite_schema WHERE type = 'table' ORDER BY tbl_name")
-            .unwrap();
-        let mut rows = tables_query.query([]).unwrap();
-        let mut expected_idx = 0;
-        while let Some(row) = rows.next().unwrap() {
-            let sql: String = row.get(0).unwrap();
+        let rows = describe_tables(&st.wallet().conn).unwrap();
+        for (actual, expected) in rows.iter().zip(expected_tables.iter()) {
```

Comment:
`zip` short-circuits, so we no longer detect whether the real database contains tables that aren't listed in `expected_tables`. I also note that we weren't detecting the reverse case previously (`expected_tables` containing tables that are not in the real database).

Simple fix is to add an assertion that the two are the same length, prior to the `zip`:
```suggestion
        assert_eq!(rows.len(), expected_tables.len());
        for (actual, expected) in rows.iter().zip(expected_tables.iter()) {
```